### ISSUE TITLE
<format>: error C2668 when import <format> as header unit and use std::format_to

### ISSUE THREAD

**Issue by lone-wolf-akela** [microsoft/STL#1919]
**Describe the bug**
When import `<format>` as a headerunit, using `std::format_to` leads to an error C2668 in Release mode.
This problem does not occur in Debug mode.

**Command-line test case**
```
D:\cpptest>type repro.cpp
import <format>;
import <string>;

int main()
{
        std::string msg;
        std::format_to(std::back_inserter(msg), "{}", "");
        return 0;
}
D:\cpptest>cl /std:c++20 /permissive- /GS /GL /W3 /Gy /Zc:wchar_t /Zi /Gm- /O2 /sdl /Zc:inline /fp:precise /D "NDEBUG" /D "_CONSOLE" /D "_UNICODE" /D "UNICODE" /WX- /Zc:forScope /Gd /Oi /MD /std:c++20  /EHsc /nologo /exportHeader /headerName:angle format string
format
string

D:\cpptest>cl /std:c++20 /permissive- /GS /GL /W3 /Gy /Zc:wchar_t /Zi /Gm- /O2 /sdl /Zc:inline /fp:precise /D "NDEBUG" /D "_CONSOLE" /D "_UNICODE" /D "UNICODE" /WX- /Zc:forScope /Gd /Oi /MD /std:c++20  /EHsc /nologo /headerUnit:angle format=format.ifc /headerUnit:angle string=string.ifc repro.cpp /c
repro.cpp
D:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.29.30035\include\format(2461): error C2668: 'abs': ambiguous call to overloaded function
D:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.29.30035\include\cstdlib(27): note: could be 'float abs(float) noexcept'
D:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.29.30035\include\cstdlib(27): note: or       'float abs(float) noexcept'
D:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.29.30035\include\format(2461): note: while trying to match the argument list '(const _Float)'
......
```

**Expected behavior**
The file should compile successfully.

**STL version**
Microsoft Visual Studio Enterprise 2019 Preview
Version 16.10.0 Preview 3.0

**Comment by fsb4000** [comment]
@StephanTLavavej Hi.
I reduced to this:

main.cpp
```
import "test.h";

int main()
{
  test(5.0);
  return 0;
}
```

test.h
```
#pragma once
#include "test2.h"

template<typename T>
auto test(T value) {
  return std::abs(value);
}
```

test2.h
```
#pragma once
extern "C" {
  int  __cdecl abs(int _X);
}


inline double abs(double _Xx){
  return 0;
}

inline float abs(float _Xx){
  return 0;
}


namespace std {
  using ::abs;
}
```

```
cl /std:c++20 /permissive- /GS /GL /W3 /Gy /Zc:wchar_t /Zi /Gm- /O2 /sdl /Zc:inline /fp:precise /D "NDEBUG" /D "_CONSOLE" /D "_UNICODE" /D "UNICODE" /WX- /Zc:forScope /Gd /Oi /MD /std:c++20  /EHsc /nologo /exportHeader  "test.h"

cl /std:c++20 /permissive- /GS /GL /W3 /Gy /Zc:wchar_t /Zi /Gm- /O2 /sdl /Zc:inline /fp:precise /D "NDEBUG" /D "_CONSOLE" /D "_UNICODE" /D "UNICODE" /WX- /Zc:forScope /Gd /Oi /MD /std:c++20  /EHsc /nologo /headerUnit "test.h=test.h.ifc" main.cpp /c

C:\Dev\STL\playground\test.h(6): error C2668: 'abs': ambiguous call to overloaded function
C:\Dev\STL\playground\test2.h(7): note: could be 'double abs(double)'
C:\Dev\STL\playground\test2.h(7): note: or       'double abs(double)'
C:\Dev\STL\playground\test.h(6): note: while trying to match the argument list '(T)'
        with
        [
            T=double
        ]
main.cpp(5): note: see reference to function template instantiation 'auto test<double>(T)' being compiled
        with
        [
            T=double
        ]
```

**Comment by StephanTLavavej** [comment]
Filed as VSO-1329224; @cdacamar has a fix.

**Comment by StephanTLavavej** [comment]
Fixed in VS 2022 17.0 Preview 2 (possibly Preview 3 but since Cameron performs the merges, I think this will get into Preview 2). Thanks again for the quick report! :smile_cat:
